x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/search/SearchApplication.java

```diff
@@ -119,14 +112,14 @@ public void writeTo(StreamOutput out) throws IOException {
                 }
             }
             @SuppressWarnings("unchecked")
-            final String[] indices = ((List<String>) params[1]).toArray(String[]::new);
+            final List<String> maybeIndices = (List<String>) params[1];
```

Comment:
Note: This method is the one check that we still do that will fail before license checks occur., if name != resourceName (see line 106). The solution is to remove the name from the request, or remove this check. Thoughts?